src/webui/api/torrentscontroller.cpp

```diff
@@ -1041,3 +1041,70 @@ void TorrentsController::categoriesAction()
 
     setResult(categories);
 }
+
+void TorrentsController::addTagsAction()
+{
+    checkParams({"hashes", "tags"});
+
+    const QStringList hashes {params()["hashes"].split('|')};
+    const QStringList tags {params()["tags"].split(',', QString::SkipEmptyParts)};
```

Comment:
I was reviewing these code recently and I think we should use `value()` method instead of `operator[]`.
```c++
const QStringList hashes {params().value("hashes").split('|')};
const QStringList tags {params().value("tags").split(',', QString::SkipEmptyParts)};
```

Reply:
Sorry about my previous note, you are free not making this change, I'm having second thought.